core/src/main/scala/io/projectglow/vcf/VariantContextToInternalRowConverter.scala

```diff
@@ -285,6 +286,48 @@ class VariantContextToInternalRowConverter(
     makeArray(strList, parseFn)
   }
 
+  private def getAnnotationArray(
+      vc: VariantContext,
+      key: String,
+      schema: StructType): Array[GenericInternalRow] = {
+    vc.getAttributeAsStringList(key, "")
+      .asScala
+      .toArray
+      .map { effect =>
+        // Providing a limit to the splitter preserves empty annotations
+        val stringAnnotations =
+          effect.split(AnnotationUtils.annotationDelimiterRegex, schema.size)
```

Comment:
Do you know if `String.split` actually uses regular expressions for one character patterns like this? If so we may be better off using Guava's `Splitter`.

Reply:
It does, at least since Java 7 - see http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/String.java#l2316.